CRM/Contact/Form/Search/Criteria.php

```diff
@@ -539,4 +541,19 @@ public static function CiviCase(&$form) {
     CRM_Case_BAO_Query::buildSearchForm($form);
   }
 
+  public static function getEnabledComponents() {
+    $apiResult = civicrm_api3('Setting', 'Get', array());
+    return $apiResult['values'][1]['enable_components'];
+  }
+
+  //the idea is to validate everything in here and remove based on possible situations
+  public static function validatePossibleComponents(&$componentModes) {
+    $enabledComponents = self::getEnabledComponents();
```

Comment:
I've done separatly because I've seen another purposes for this use. At least I think we should use an auxiliar variable (like ```$enabledComponents```) to not calling the function more than once.